src/components/Dashboard.tsx

```diff
@@ -1,46 +1,90 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react'
+import logo from '../../public/images/logo.PNG'
 import Image from 'next/image';
-import { UserData, ComponentProps } from '@/types/types';
-import { getTodaysDate } from '@/utils';
+import { getTodaysDate, getZodiacSign } from '@/utils';
 import Navbar from '../components/Navbar';
 import { insights } from '@/mockdata';
 import Router from "next/router";
-import Logo from '@/components/logo';
+import { Passage } from '@passageidentity/passage-js';
+import { PassageUserInfo } from '@passageidentity/passage-elements/passage-user';
+import dotenv from 'dotenv';
+import { AuthProps } from '@/types/types';
 
+const getCurrentUser = async () => {
+  dotenv.config();
+  const passageAppId = process.env.NEXT_PUBLIC_PASSAGE_APP_ID;
 
-type DashboardProps = ComponentProps & {
-  logOut: () => void
-};
+  if (!passageAppId) {
+    throw new Error('NEXT_PUBLIC_PASSAGE_APP_ID is not defined in the environment.');
+  }
 
+  const passage = new Passage(passageAppId);
+  const user = passage.getCurrentUser();
+  const userInfo = await user.userInfo();
+  return userInfo
+}
 
-export default function Dashboard({ isAuthorized, data, logOut}: DashboardProps) {
-  const [user, setUser] = useState<UserData | null>(null)
-  const [userInsights, setUserInsights] = useState(insights)
-
+export default function Dashboard({ isAuthorized, userID }: AuthProps) {
+  const [user, setUser] = useState<PassageUserInfo | undefined>(undefined);
+  // const [userInsights, setUserInsights] = useState(insights)
 
   useEffect(() => {
     if (!isAuthorized) {
       Router.push("/");
-    }      
-  
-    if (data) setUser(data[0])
-    //api call needs to be made to get horoscope
+    } else {
+      getCurrentUser()
+        .then((data) => {
+          setUser(data)
+          checkForUser(userID)
+        })
+        .catch((error) => {
+          console.error('Error fetching user:', error);
+        });
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [isAuthorized]);
-  
+
+  const checkForUser = async (userID: string | number) => {
+    const res = await fetch('/api/getUsers')
+    .then((res) => res.json())
+    .then((allIDs) => {
+      if (!allIDs.includes(userID) && user) {
+        addNewUser(user)
+      }
+    })
+  }
```

Comment:
I think these are missing error handling in a .catch?

Reply:
ahh good _catch_, I will add .catch error handling!